src/code/PreRender.c

```diff
@@ -1,42 +1,615 @@
+
+    x2 = 0;
+    x = this->height;
+    add = 0x1000 / (this->width*2);
+
+    while (x > 0)
+    {
+        lrs = this->width-1;
+
+        if (x < add) add = x;
+            
+        uls = 0;
+        ult = x2;
+        lrt = (x2 + add-1);
+
+        gDPLoadTextureTile(gfx++, fbuf, G_IM_FMT_RGBA, G_IM_SIZ_16b, this->width, this->height, uls, ult, lrs, lrt, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+        if (1) { }
+        gSPTextureRectangle(gfx++, uls<<2, ult<<2, (lrs+1)<<2, (lrt+1)<<2, G_TX_RENDERTILE, uls<<5, ult<<5, 1<<10, 1<<10);
+
+        x -= add;
+        x2 += add;
+    }
+
+    gDPPipeSync(gfx++);
+    gDPSetColorImage(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_16b, this->width, this->fbuf);
+    *gfxp = gfx;
+}
+
+void func_800C1AE8(PreRenderContext* this, Gfx** gfxp, void* fbuf, void* fbufSave)
+{
+    func_800C170C(this, gfxp, fbuf, fbufSave, 0xFF, 0xFF, 0xFF, 0xFF);
+}
+
+void func_800C1B24(PreRenderContext* this, Gfx** gfxp, void* fbuf, void* cvgSave)
+{
+    Gfx* gfx;
+    s32 x;
+    s32 x2;
+    s32 add;
+    s32 uls;
+    s32 ult;
+    s32 lrs;
+    s32 lrt;
+
+    LogUtils_CheckNullPointer("this", this, "../PreRender.c", 422);
+    LogUtils_CheckNullPointer("glistpp", gfxp, "../PreRender.c", 423);
+    gfx = *gfxp;
+    LogUtils_CheckNullPointer("glistp", gfx, "../PreRender.c", 425);
+
+    gDPPipeSync(gfx++);
+    gDPSetOtherMode(gfx++, G_AD_DISABLE | G_CD_DISABLE | G_CK_NONE | G_TC_FILT | G_TF_POINT | G_TT_NONE | G_TL_TILE | G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE, G_AC_NONE | G_ZS_PRIM | G_RM_PASS | G_RM_OPA_CI2);
+    gDPSetCombineLERP(gfx++, 0,0,0,TEXEL0, 0,0,0,0, 0,0,0,TEXEL0, 0,0,0,0);
+    gDPSetColorImage(gfx++, G_IM_FMT_I, G_IM_SIZ_8b, this->width, cvgSave);
+    gDPSetScissor(gfx++, G_SC_NON_INTERLACE, 0, 0, this->width, this->height);
+
+    x = this->height;
+    x2 = 0;
+    add = 0x1000 / (this->width*2);
+    
+    while (x > 0)
+    {
+        lrs = this->width-1;
+        if (x < add) add = x;
+        uls = 0;
+        ult = x2;
+        lrt = (x2+add)-1;
+            
+        gDPLoadTextureTile(gfx++, fbuf, G_IM_FMT_IA, G_IM_SIZ_16b, this->width, this->height, uls, ult, lrs, lrt, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+        if (1) { }
+        gSPTextureRectangle(gfx++, uls<<2, ult<<2, (lrs+1)<<2, (lrt+1)<<2, G_TX_RENDERTILE, uls<<5, ult<<5, 1<<10, 1<<10);
+
+        x -= add;
+        x2 += add;
+    }
+
+    gDPPipeSync(gfx++);
+    gDPSetColorImage(gfx++, G_IM_FMT_RGBA, G_IM_SIZ_16b, this->width, this->fbuf);
+    *gfxp = gfx;
+}
+
+void func_800C1E9C(PreRenderContext* this, Gfx** gfxp)
+{
+    LogUtils_CheckNullPointer("this->zbuf_save", this->zbufSave, "../PreRender.c", 481);
+    LogUtils_CheckNullPointer("this->zbuf", this->zbuf, "../PreRender.c", 482);
+
+    if (!this->zbufSave || !this->zbuf)
+        return;
```

Comment:
here and below there are some braces omitted on single line ifs which we dont do